packages/date-picker/src/DatePicker/DatePicker.spec.tsx

```diff
@@ -141,31 +142,28 @@ describe("<DatePicker /> - Focus element", () => {
     it("shows focus on input and opens the calendar", () => {
       render(<DatePickerWrapper selectedDay={new Date("2022-03-1")} />)
 
-      expect(screen.queryByRole("dialog")).not.toBeInTheDocument()
-
-      const input = screen.getByLabelText("Input label", { selector: "input" })
+      const input = screen.getByRole("combobox", { name: "Input label" })
       userEvent.click(input)
 
       waitFor(() => {
-        expect(screen.queryByRole("dialog")).toBeInTheDocument()
+        expect(screen.queryByRole("dialog")).toBeVisible()
         expect(input).toHaveFocus()
       })
     })
 
     it("returns focus to the input when the user clicks a valid day on the calendar", () => {
       render(<DatePickerWrapper selectedDay={new Date("2022-03-1")} />)
 
-      expect(screen.queryByRole("dialog")).not.toBeInTheDocument()
-
-      const input = screen.getByLabelText("Input label", { selector: "input" })
+      const input = screen.getByRole("combobox", { name: "Input label" })
       userEvent.click(input)
 
       waitFor(() => {
-        expect(screen.queryByRole("dialog")).toBeInTheDocument()
+        expect(screen.queryByRole("dialog")).toBeVisible()
       })
 
-      const dateToSelect = screen.getByText("6th March (Sunday)").parentElement
-      if (dateToSelect) userEvent.click(dateToSelect)
+      const dateToSelect = screen.getByText("6th March (Sunday)").parentElement!
```

Comment:
You're being quite inconsistent in your changes?

If `const dateToSelect = screen.getByRole("button", { name: todayFormatted })` is the pattern you recommend, then it's best to use it consistently.